### PR TITLE
tools: adding some version notes alphabetization checks to check_format

### PR DESCRIPTION
**Pull request by alyssawilk**
Risk Level: n/a 
Testing: manual as seen by new release notes
Docs Changes: n/a
Release Notes: sort of...

### CODE REVIEW

**Comment by mattklein123** [comment]
Yes amazing!!! Can you merge master?

File: tools/code_format/check_format.py [hunk 1]
```diff
@@ -415,9 +417,26 @@ def reportError(message):
...
-                  "Does not match VERSION_HISTORY_NEW_LINE_REGEX in check_format.py\n %s" % line)
+    # Do basic alphabetization checks of the first word on the line and the
+    # first word after the :
+    if line.startswith("*"):
+      match = VERSION_HISTORY_NEW_LINE_REGEX.match(line)
+      if not match:
+        reportError("Version history line malformed. "
+                    "Does not match VERSION_HISTORY_NEW_LINE_REGEX in check_format.py\n %s" % line)
+      else:
+        first_word = match.groups()[0]
```
Comment (asraa) [inline tools/code_format/check_format.py:428]: Based on the * quantifier in `VERSION_HISTORY_NEW_LINE_REGEX` could this just be `:`? Is that OK? (Alphabetization makes sense of it, but just as a release note)

Comment (alyssawilk) [reply tools/code_format/check_format.py:428]: I guess arguably - that'd be a useless release note but we should complain cleanly :-)

I'm doing a follow-up to make sure sentences end with . - OK if I change the * to + there, just for release reasons?

Comment (asraa) [reply tools/code_format/check_format.py:428]: Yes, that seems like a better place! LGTM otherwise

Comment (mattklein123) [reply tools/code_format/check_format.py:428]: OK sounds good I will force merge once the docs build passes.

**Pull request merged**